corehq/apps/userreports/expressions/specs.py

```diff
@@ -239,3 +240,27 @@ def __call__(self, item, context=None):
         for property_name, expression in self._compiled_properties.items():
             ret[property_name] = expression(item, context)
         return ret
+
+
+class EvalExpressionSpec(JsonObject):
+    type = TypeProperty('evaluator')
+    statement = StringProperty(required=True)
+    context_variables = DictProperty(required=True)
+
+    def configure(self, context_variables):
+        self._context_variables = context_variables
+
+    def __call__(self, item, context=None):
+        var_dict = self.get_variables(item, context)
+        return eval_statements(self.statement, var_dict)
+
+    def get_variables(self, item, context):
+        var_dict = {
+            slug: variable_expression(item, context)
+            for slug, variable_expression in self._context_variables.items()
+        }
+        var_types = set(type(value) for value in var_dict.values())
+        if not var_types.issubset(set([int, float, long])):
+            raise BadSpecError
```

Comment:
I don't think this is the right error to raise, as it's potentially an issue with the data, not the spec